docs/building-a-form/about-the-json-schema-standard-and-react-jsonschema-form.md

```diff
@@ -0,0 +1,298 @@
+- A *widget* is the html input element that accepts data from the user. The schemaform uses `text`, `email`, `checkbox`, `radio`, `select`, and `textarea`. While there are many widgets provided by rjsf, the defaults are overwritten with these versions.
+
+##### Example schema
+
+```
+{
+  type: 'string'
+}
+```
+
+##### Resulting hierarchy
+
+The two `Field` components determine which fields and widgets to render. `SchemaField` uses the two schemas the library accepts, `schema` and `uiSchema`, to determine what other `Field` component to render. The example chose `StringField` because the schema type was `string`. The `StringField` component then rendered `TextWidget`, based on `schema` and `uiSchema`, because the only information provided was that the field is a string (the default widget type).
+
+```
+<SchemaField>
+  <StringField>
+    <FieldTemplate>
+      <TextWidget/>
+    </FieldTemplate>
+  </StringField>
+</SchemaField>
+```
+
+##### Example schema
+
+```
+{
+  type: 'string',
+  enum: ['first', 'second', 'third']
+}
+```
+
+##### Resulting hierarchy
+
+The hierarchy for this field uses `SelectWidget` instead of `TextWidget`, because `StringField` detected the `enum` property in the schema.
+
+```
+<SchemaField>
+  <StringField>
+    <FieldTemplate>
+      <SelectWidget/>
+    </FieldTemplate>
+  </StringField>
+</SchemaField>
+```
+
+To override rules in `uiSchema`, specify a `ui:widget` property set to `text`, `email`, `checkbox`, or your own custom widget. The `ui:field` property can specify a specific or custom field.
+
+For `object` and `array` schema types, a field component does something other than figure out what widgets to render.
+
+##### Example schema: `object`
+
+This is an `object` schema with two string fields.
+
+```
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    },
+    field2: {
+      type: 'string'
+    }
+  }
+}
+```
+##### Resulting hierarchy
+
+The `ObjectField` component renders a `SchemaField` component for each of its properties. Those properties are both `string` types, so it looks like the first hierarchy, but nested.
+
+```
+<SchemaField>
+  <ObjectField>
+    <SchemaField>
+      <StringField>
+        <FieldTemplate>
+          <TextWidget/>
+        </FieldTemplate>
+      </StringField>
+    </SchemaField>
+    <SchemaField>
+      <StringField>
+        <FieldTemplate>
+          <TextWidget/>
+        </FieldTemplate>
+      </StringField>
+    </SchemaField>
+  </ObjectField>
+</SchemaField>
+```
+
+##### Example schema: `ArrayField`
+
+`ArrayField` renders a `SchemaField` component for each item in the array. The library only uses the array field where each item is an object type schema.
+
+```
+{
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      field1: {
+        type: 'string'
+      },
+      field2: {
+        type: 'string'
+      }
+    }
+  }
+}
+```
+
+##### Field components
```

Comment:
Perhaps rename to `List of field components`, as that's really what this section is?